targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_MCU_K64F/mbed_crc_api.c

```diff
@@ -0,0 +1,65 @@
+#include "crc_api.h"
+
+#include "drivers/fsl_crc.h"
+#include "platform/mbed_assert.h"
+
+#ifdef DEVICE_CRC
+
+static crc_bits_t width;
+
+bool hal_crc_is_supported(const crc_mbed_config_t* config)
+{
+  if (config == NULL)
+    return false;
+
+  if ((config->polynomial & 0x80008000) == 0)
+    return false;
+
+  return true;
+}
+
+void hal_crc_compute_partial_start(const crc_mbed_config_t* config)
+{
+  if (config == NULL)
+    return;
+
+  width = ((config->polynomial & 0xFFFF0000U) != 0) ? kCrcBits32 : kCrcBits16;
+
+  crc_config_t platform_config;
+  platform_config.polynomial         = config->polynomial;
+  platform_config.seed               = config->initial_xor;
+  platform_config.reflectIn          = config->reflect_in;
+  platform_config.reflectOut         = config->reflect_out;
+  platform_config.complementChecksum = true;
```

Comment:
If given `final_xor` config value is equal to `all ones` (e.q. 0xFFFF_FFFF for 32-bit CRC), then `platform_config.complementChecksum` should be set to `true`. 
If given `final_xor` config value is 0, then `platform_config.complementChecksum` should be set to `false`.
If given `final_xor` config value is different than both mentioned values, then `platform_config.complementChecksum` should be set to `false`, software `xor` operation should be performed on the hardware result and the result of the operation should be returned.

Reply:
Ah okay, I assumed it wasn't supported in the hardware but that will have the same effect.

Reply:
Complement checksum is different name for final XOR. Its same parameter but different hardware modules accept either true or full value. It should not be separate argument from HAL.